src/test/regress/sql/gporca.sql

```diff
@@ -3341,6 +3341,74 @@ with cte as (
 select * 
 from empty_cte_tl_test
 where id in(select id from cte);
+
+-- check that query which hangs before patch https://github.com/greenplum-db/gpdb/pull/14896 (1),
+-- and fallbacks to postgres due to patch https://github.com/arenadata/gpdb/pull/302 above (1),
+-- now doesn't fallback
+
+-- start_matchsubs
+-- m/Executor [mM]emory: .*/
+-- s/Executor [mM]emory: .*/Executor Memory: /
+-- m/Extra Text:.*/
+-- s/Extra Text:.*/Extra Text:/
+-- m/Memory used:\s+[0-9]+kB.*/
+-- s/Memory used:\s+[0-9]+kB.*/Memory used:/
+-- m/\(actual time=\d+\.\d+..\d+\.\d+ rows=\d+ loops=\d+\)/
+-- s/\(actual time=\d+\.\d+..\d+\.\d+ rows=\d+ loops=\d+\)/(actual time=##.###..##.### rows=# loops=#)/
+-- m/Execution time: \d+\.\d+ ms/
+-- s/Execution time: \d+\.\d+ ms/Execution time: ##.### ms/
+-- m/Planning time: \d+\.\d+ ms/
+-- s/Planning time: \d+\.\d+ ms/Planning time: ##.### ms/
+-- end_matchsubs
+
+DROP TABLE tab1, tab2;
+CREATE TABLE tab1(dict_id INT, tab_1_join_col INT);
+CREATE TABLE tab2(part_column INT, tab_1_join_col INT);
+CREATE TABLE dict_tab(dict_id INT PRIMARY KEY) DISTRIBUTED REPLICATED;
+
+EXPLAIN (ANALYZE, COSTS OFF) WITH
+	cte1 AS (SELECT dict_id, tab_1_join_col FROM tab1),
+	cte2 AS (
+		SELECT a.part_column, e.dict_id
+		FROM tab2 a
+		LEFT JOIN cte1 AS e
+			ON e.tab_1_join_col = a.tab_1_join_col
+		LEFT JOIN cte1 AS e1
+			ON e1.tab_1_join_col = a.tab_1_join_col
+)
+SELECT t.*, ROW_NUMBER() OVER(PARTITION BY t.part_column) AS row_num
+FROM cte2 AS t LEFT JOIN dict_tab tp2 ON tp2.dict_id = t.dict_id;
+
+DROP TABLE tab1, tab2, dict_tab;
+
+-- query which raise a segfault before patch https://github.com/greenplum-db/gpdb/pull/14896 (1),
+-- and fallbacks to postgres after applyng patch https://github.com/arenadata/gpdb/pull/302 above (1),
+-- now doesn't fallback
+CREATE TABLE tab1(dict_id INT, tab_1_join_col INT);
+INSERT INTO tab1 VALUES(1, 1), (1, 2), (1, 3);
+CREATE TABLE tab2(part_column INT, tab_1_join_col INT);
+INSERT INTO tab2 VALUES (1, 1), (2, 2), (3, 3), (1, 4);
+CREATE TABLE dict_tab(dict_id INT PRIMARY KEY) DISTRIBUTED REPLICATED;
+INSERT INTO dict_tab SELECT i FROM generate_series(1, 1000) i;
+ANALYZE dict_tab;
+
+EXPLAIN (ANALYZE, COSTS OFF) WITH cte1 AS (
```

Comment:
Do we really need to fulfill the request?

Reply:
You are changing the planning while you are checking the execution!

Reply:
I think that errors when executing a request should be checked separately and in a separate patch.
Therefore, in this patch, it makes no sense to check the execution of the request.

Reply:
This tests compares that the plans were generated by ORCA (there was no fallback) and the plan generated by ORCA won't hang/segfault. IMO the both checks may be implemented in this patch